servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/QueueDynamicCompositeCancellable.java

```diff
@@ -55,12 +50,14 @@ public void cancel() {
 
     @Override
     public boolean add(Cancellable toAdd) {
-        final Queue<Cancellable> currentQueue = queue;
-        if (currentQueue == null || !currentQueue.offer(toAdd)) {
+        if (counter < 0 || !queue.offer(toAdd)) { // make best effort to avoid the queue if cancelled already
```

Comment:
nit: The CLQ is unbounded so the check for false, is not needed. I would remove it for logic simplification.
Maybe with a comment in the queue declaration in case we ever change that.

Reply:
this has been updated to `ConcurrentStack` and the logic has been simplified accordingly.